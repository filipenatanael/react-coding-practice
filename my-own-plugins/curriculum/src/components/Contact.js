import React from 'react';

const Contact = ({ props }) => (
  <div className="contact-container container-block">
      <ul className="list-unstyled contact-list">
          <li className="email"><i className="fas fa-envelope"></i><a href="mailto: yourname@email.com">alan.doe@website.com</a></li>
          <li className="phone"><i className="fas fa-phone"></i><a href="tel:0123 456 789">0123 456 789</a></li>
          <li className="website"><i className="fas fa-globe"></i><a href="https://themes.3rdwavemedia.com/bootstrap-templates/resume/orbit-free-resume-cv-bootstrap-theme-for-developers/" target="_blank">portfoliosite.com</a></li>
          <li className="linkedin"><i className="fab fa-linkedin-in"></i><a href="#" target="_blank">linkedin.com/in/alandoe</a></li>
          <li className="github"><i className="fab fa-github"></i><a href="#" target="_blank">github.com/username</a></li>
          <li className="twitter"><i className="fab fa-twitter"></i><a href="https://twitter.com/3rdwave_themes" target="_blank">@twittername</a></li>
      </ul>
  </div>
);

export default Contact;
