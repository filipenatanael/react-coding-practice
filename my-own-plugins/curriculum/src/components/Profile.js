import React from 'react';

const Profile = ({ props }) => (
  <div className="profile-container">
      <img className="profile" src="https://themes.3rdwavemedia.com/demo/orbit/assets/images/profile.png" alt="" />
      <h1 className="name">Alan Doe</h1>
      <h3 className="tagline">Full Stack Developer</h3>
  </div>
);

export default Profile;
