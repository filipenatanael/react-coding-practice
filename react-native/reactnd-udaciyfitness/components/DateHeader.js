import React, { Component } from 'react';
import { Text } from 'react-native';

export default function DateHeader({ date }) {
  return (
    <Text style={{ color: '#fff' }}>
      {date}
    </Text>
  )
}
