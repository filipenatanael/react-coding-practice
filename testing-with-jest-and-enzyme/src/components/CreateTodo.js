import React, { Component } from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'
import {connect} from 'react-redux'
import {createTodo} from '../redux/modules/todo/actions'


export class CreateTodo extends Component {
  constructor() {
    super();
    this.state = {
      todo: {
        title: null,
        project: null
      },
      formOpen: false
    }

    this.handleOpen=this.handleOpen.bind(this);
    this.handleClose=this.handleClose.bind(this);
    this.handleFieldChange=this.handleFieldChange.bind(this);
    this.handleCreate=this.handleCreate.bind(this);
  }

  handleOpen() {
    this.setState({formOpen: true});
  }

  handleClose() {
    this.setState({todo: {title: null, project: null}});
    this.setState({formOpen: false});
  }

  handleFieldChange(e) {
    var field = e.target.name;
    var value = e.target.value;
    var todo = this.state.todo;
    todo = Object.assign({}, todo, {[field]: value});
    this.setState({todo: todo});
  }

  handleCreate() {
    this.props.createTodo(this.state.todo);
    this.handleClose();
  }

  render() {
    return (
      <div className="column" style={{textAlign: 'center'}}>
        {this.state.formOpen ?
          <div className="ui centered grid">
            <div className="eight wide column" style={{boxShadow: '0 0 0 1px #d4d4d5, 0 2px 0 0 #a5673f, 0 1px 3px 0 #d4d4d5', margin: '20px 0', padding: 0}}>
              <div className="header" style={{display: 'block',
    fontFamily: 'Lato,"Helvetica Neue",Arial,Helvetica,sans-serif',
    background: '#fff',
    margin: 0,
    padding: '1.25rem 1.5rem',
    boxShadow: 'none',
    color: 'rgba(0,0,0,.85)',
    borderBottom: '1px solid rgba(34,36,38,.15)',fontSize: '1.42857143rem',
    lineHeight: '1.28571429em',
    fontWeight: 700}}>Create Todo</div>
              <div className="content" style={{display: 'block',
    width: '100%',
    fontSize: '1em',
    lineHeight: 1.4,
    padding: '1.5rem',
    background: '#fff'}}>
                <div className='ui form'>
                  <div className='field'>
                    <label>Title</label >
                    <input type='text' name="title" onChange={this.handleFieldChange} />
                  </div>
                  <div className='field'>
                    <label>Project</label>
                    <input type='text' name="project" onChange={this.handleFieldChange} />
                  </div>
                </div>
              </div>
              <div className="actions" style={{background: '#f9fafb',
    padding: '1rem 1rem',
    borderTop: '1px solid rgba(34,36,38,.15)',
    textAlign: 'right'}}>
                <button className='ui inverted green button' onClick={this.handleCreate}>
                  <i className='checkmark icon'></i> Create
                </button>
                <button className='ui inverted red button' onClick={this.handleClose}>
                  <i className='remove icon'></i> Cancel
                </button>
              </div>
            </div>
          </div>
         :
          <div className="ui icon button" data-tooltip="Create New Todo" data-position="bottom center" onClick={this.handleOpen}>
            <i className="add icon"></i>
          </div>
        }
      </div>
    )
  }
}

CreateTodo.propTypes = {
  createTodo: PropTypes.func.isRequired,
};

export default connect(null, {createTodo})(CreateTodo);
