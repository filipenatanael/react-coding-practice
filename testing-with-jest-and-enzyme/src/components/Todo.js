import React, { Component } from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'
import {connect} from 'react-redux'
import {editTodo, toggleTodo, deleteTodo} from '../redux/modules/todo/actions'

export class Todo extends Component {
  constructor() {
    super();
    this.state = {
      formOpen: false,
      todo: {}
    }
    this.handleOpen=this.handleOpen.bind(this);
    this.handleClose=this.handleClose.bind(this);
    this.handleFieldChange=this.handleFieldChange.bind(this);
    this.handleEdit=this.handleEdit.bind(this);
    this.handleDelete=this.handleDelete.bind(this);
    this.handleToggle=this.handleToggle.bind(this);
  }

  handleOpen() {
    this.setState({formOpen: true});
  }

  handleClose() {
    this.setState({formOpen: false});
    this.setState({todo: {}});
  }

  handleFieldChange(e) {
    var field = e.target.name;
    var value = e.target.value;
    var todo = this.state.todo;
    var todo = Object.assign({}, todo, {[field]: value});
    this.setState({todo: todo});
  }

  handleEdit() {
    this.props.editTodo(this.props.id, this.state.todo);
    this.handleClose();
  }

  handleDelete() {
    this.props.deleteTodo(this.props.id);
  }

  handleToggle() {
    this.props.toggleTodo(this.props.id, {done: !this.props.done})
  }

  render() {
    return (
      <div className="column">
        <div className="ui brown card">
          <img className="ui image" src={this.props.url} />
          {this.state.formOpen ?
            <div className="content">
              <div className='ui form'>
                <div className='field'>
                  <label>Title</label>
                  <input type='text'
                    name="title"
                    defaultValue={this.props.title}
                    onChange={this.handleFieldChange}
                  />
                </div>
                <div className='field'>
                  <label>Project</label>
                  <input type='text'
                    name="project"
                    defaultValue={this.props.project}
                    onChange={this.handleFieldChange}
                  />
                </div>
              </div>
            </div> :
            <div className="content">
              <div className="header">{this.props.title}</div>
              <div className="meta">{this.props.project}</div>
              <div className="meta">Created {moment(this.props.createdAt).fromNow()}</div>
            </div>
          }
          <div className="extra content">
            {this.state.formOpen ?
              <div className="ui two buttons">
                <button className='ui basic green button' onClick={this.handleEdit}>
                  <i className='checkmark icon'></i> Update
                </button>
                <button className='ui basic red button' onClick={this.handleClose}>
                  <i className='remove icon'></i> Cancel
                </button>
              </div> :
              <div>
                <div className="ui toggle checkbox" style={{marginBottom: '10px'}}>
                  <input type="checkbox" name="public" value="on" defaultChecked ={this.props.done} onChange={this.handleToggle}/>
                  <label>Complete</label>
                </div>
                <div className="ui two buttons">
                  <button className='ui basic green button' onClick={this.handleOpen}>Edit</button>
                  <button className="ui basic red button" onClick={this.handleDelete}>Delete</button>
                </div>
              </div>
            }
          </div>
        </div>
      </div>
    )
  }
}

Todo.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  project: PropTypes.string.isRequired,
  done: PropTypes.bool.isRequired,
  url: PropTypes.string.isRequired,
  createdAt: PropTypes.string.isRequired,
  editTodo: PropTypes.func.isRequired,
  toggleTodo: PropTypes.func.isRequired,
  deleteTodo: PropTypes.func.isRequired
};

export default connect(null, {editTodo, toggleTodo, deleteTodo})(Todo);
