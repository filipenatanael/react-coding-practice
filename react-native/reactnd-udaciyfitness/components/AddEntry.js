import React, { Component } from 'react';
import { View, Text, TouchableOpacity, StyleSheet, Platform } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { getMetricMetaInfo, timeToString, getDailyReminderValue } from '../utils/helpers';
import { submitEntry, removeEntry } from '../utils/api';
import { purple, white } from '../utils/colors'

import { connect } from 'react-redux';
import { addEntry } from '../actions';

import UdaciSlider from './UdaciSlider';
import UdaciSteppers from './UdaciSteppers';
import DateHeader from './DateHeader';
import TextButton from './TextButton';

function SubmitBtn ({ onPress }) {
  return (
    <TouchableOpacity
      style={Platform.OS === 'ios' ? styles.iosSubmitBtn : styles.androidSubmitBtn}
      onPress={onPress}>
        <Text style={styles.submitBtnText}>Submit</Text>
    </TouchableOpacity>
  )
}

class AddEntry extends Component {
  state = {
    run: 0,
    bike: 0,
    swim: 0,
    sleep: 0,
    eat: 0,
  }

  increment = (metric) => {
    const { max, step } = getMetricMetaInfo(metric)

    this.setState((state) => {
      const count = state[metric] + step

      return {
        ...state,
        [metric]: count > max ? max : count
      }
    })
  }

  decrement = (metric) => {
    this.setState((state) => {
      const count = state[metric] - getMetricMetaInfo(metric).step

      return {
        ...state,
        [metric]: count < 0 ? 0 : count
      }
    })
  }

  slide = (metric, value) => {
    this.setState(() => ({
      [metric]: value
    }))
  }

  onSubmit = () => {
    const key = timeToString()
    const entry = this.state

    this.props.dispatch(addEntry({
      [key]: entry
    }))

    // Update Redux
    this.setState(() => ({ run: 0, bike: 0, swim: 0, sleep: 0, eat: 0 }))

    // Navigate to Home

    // Save to 'DB'
    submitEntry({ key, entry })

    // Clearn local notification
  }

  reset = () => {
    const key = timeToString()

    this.props.dispatch(addEntry({
      [key]: getDailyReminderValue()
    }))

    // Update redux

    // Route to Home

    // Update 'DB'
    removeEntry(key)
  }

  render() {
    const metaInfo = getMetricMetaInfo()

    if (this.props.alreadyLogged) {
      return (
        <View style={styles.center}>
          <Ionicons
            name={Platform.OS === 'ios' ? 'ios-happy-outline' : 'md-happy'}
            size={100} />

          <Text>You already logged your information for today.</Text>
          <TextButton style={{ padding: 10 }} onPress={this.reset}>
            Reset
          </TextButton>
        </View>
      )
    }

    return (
      <View style={styles.container}>
        <DateHeader date={(new Date()).toLocaleDateString()} />
        <Text>{JSON.stringify(this.state)}</Text>

        {Object.keys(metaInfo).map((key) => {
          const { getIcon, type, ...rest } = metaInfo[key] // metaInfo[run]...
          const value = this.state[key]

          return (
            <View key={key} style={styles.row}>
              {getIcon()}

              {type === 'slider'
                ? <UdaciSlider
                    value={value}
                    onChange={(value) => this.slide(key, value)}
                    {...rest} />
                : <UdaciSteppers
                    value={value}
                    {...rest}
                    onIncrement={() => this.increment(key)}
                    onDecrement={() => this.decrement(key)} />
              }
            </View>
          )
        })}

        <SubmitBtn onPress={this.onSubmit} />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 20,
    backgroundColor: 'white'
  },
  row: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center'
  },
  iosSubmitBtn: {
    backgroundColor: purple,
    padding: 10,
    borderRadius: 7,
    height: 45,
    marginLeft: 40,
    marginRight: 40
  },
  androidSubmitBtn: {
    backgroundColor: purple,
    padding: 10,
    paddingLeft: 30,
    paddingRight: 30,
    height: 45,
    borderRadius: 2,
    alignSelf: 'flex-end',
    justifyContent: 'center',
    alignItems: 'center'
  },
  submitBtnText: {
    color: 'white',
    fontSize: 22,
    textAlign: 'center'
  },
  center: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    marginLeft: 30,
    marginRight: 30,
  }
});

function mapStateToProps (state) {
  const key = timeToString() // Today: 2017-12-19

  return {
    alreadyLogged: state[key] && typeof state[key].today === 'undefined'
  }
}

export default connect(mapStateToProps)(AddEntry)
