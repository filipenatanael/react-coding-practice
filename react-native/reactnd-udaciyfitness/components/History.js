import React, { Component } from 'react';
import { View, Text, TouchableOpacity, StyleSheet, Platform } from 'react-native';
import { connect } from 'react-redux';
import UdaciFitnessCalendar from 'udacifitness-calendar';
import { AppLoading } from 'expo';
import { receiveEntries, addEntry } from '../actions';
import { timeToString, getDailyReminderValue } from '../utils/helpers';
import { fetchCalendarResults } from '../utils/api';
import DateHeader from './DateHeader';
import MetricCard from './MetricCard';

class History extends Component {
  state = {
    ready: false
  }

  componentDidMount() {
    const { dispatch } = this.props;

    fetchCalendarResults()
      .then((entries) => dispatch(receiveEntries(entries)))
      .then(({ entries }) => {
        if (!entries[timeToString()]) {
          // Don't forget to log your data today!
          dispatch(addEntry({
            [timeToString()]: getDailyReminderValue()
          }));
        }
      }).then(() => this.setState(() => ({
          ready: true
      })));
  }

  renderItem = ({ today, ...metrics }, formattedDate, key) => (
    <View style={styles.item}>
      {today
        ? <View>
            <DateHeader
              date={formattedDate}
            />
            <Text style={styles.noDataText}>
              {today}
            </Text>
          </View>
        : <TouchableOpacity
            onPress={() => console.log('Pressed!')}
          >
            <MetricCard date={formattedDate} metrics={metrics} />
          </TouchableOpacity>
      }
    </View>
  )

  renderEmptyDate(formattedDate) {
    return (
      <View>
        <Text>{JSON.stringify(this.props)}</Text>
        <Text>No Data for this day</Text>
      </View>
    );
  }

  render() {
    const { entries } = this.props;
    const { ready } = this.state;

    if (ready === false) {
      return <AppLoading />;
    }

    return (
        <UdaciFitnessCalendar
          items={entries}
          renderItem={this.renderItem}
          renderEmptyDate={this.renderEmptyDate}
        />
    );
  }
}

const styles = StyleSheet.create({
  item: {
    backgroundColor: '#fff',
    borderRadius: Platform.OS === 'ios' ? 16 : 2,
    padding: 20,
    marginLeft: 10,
    marginRight: 10,
    marginTop: 17,
    justifyContent: 'center',
    shadowRadius: 3,
    shadowOpacity: 0.8,
    shadowColor: 'rgba(0,0,0,0.24)',
    shadowOffset: {
      width: 0,
      height: 3
    }
  }
});

function mapStateToProps(entries) {
  return {
    entries
  };
}

export default connect(mapStateToProps)(History);
