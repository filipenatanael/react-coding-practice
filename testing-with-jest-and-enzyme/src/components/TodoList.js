import React, { Component } from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'
import {connect} from 'react-redux';
import {fetchTodos} from '../redux/modules/todo/actions'
import Todo from './Todo'
import CreateTodo from './CreateTodo'

export class TodoList extends Component {
  constructor() {
    super();
  }

  componentWillMount() {
    this.props.fetchTodos();
  }

  render() {
    return (
      <div>
        <div className="ui inverted segment" style={{borderRadius: '0px'}}>
          <h1 className="ui grey inverted center aligned header">
            <img className="ui large centered image" src='../images/React-icon.svg' />
            <div className="content">React Testing</div>
          </h1>
        </div>
        {this.props.todo.requesting &&
          <div className="ui centered container one column grid" style={{marginTop: '30px', marginBottom: '20px'}}>
            <div className="column">
              <div className="ui active loader big"></div>
            </div>
          </div>
        }
        <div className="ui centered container one column grid">
          <CreateTodo />
        </div>
        { !this.props.todo.requesting ?
          <div className="ui centered container three column grid">
            {this.props.todo.todos.length ?
                this.props.todo.todos.map(todo => {
                  return (
                    <Todo key={todo.id}
                      id={todo.id}
                      title={todo.title}
                      project={todo.project}
                      done={todo.done}
                      url={todo.image}
                      createdAt={todo.createdAt}
                    />
                  )
                })
                :
                <div className="Column" style={{fontSize: 'xx-large'}}>No Todos</div>
            }
          </div> :
          null
        }
      </div>
    )
  }
}


TodoList.propTypes = {
  todo: PropTypes.object.isRequired,
  fetchTodos: PropTypes.func.isRequired
};

export default connect(
  state => ({
    todo: state.todo
  }),
  {fetchTodos}
)(TodoList);