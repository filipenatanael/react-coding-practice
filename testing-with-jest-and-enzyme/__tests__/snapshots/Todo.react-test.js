// Link.react-test.js
import React from 'react';
import {Todo} from '../../src/components/Todo';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import to_json from 'enzyme-to-json';
import moment from 'moment';

it('Renders correctly', () => {
  const component = renderer.create(
    <Todo
      id = '1'
      title = 'Todo'
      project = 'Project'
      done = {false}
      url = 'https://www.testurl.com/images/image'
      createdAt = {moment().subtract(1, 'days').format()}
      editTodo = {jest.fn()}
      toggleTodo = {jest.fn()}
      deleteTodo = {jest.fn()}
    ></Todo>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('Clicking edit button brings up edit form', () => {
  const wrapper = shallow(
    <Todo
      id = '1'
      title = 'Todo'
      project = 'Project'
      done = {false}
      url = 'https://www.testurl.com/images/image'
      createdAt = {moment().subtract(1, 'days').format()}
      editTodo = {jest.fn()}
      toggleTodo = {jest.fn()}
      deleteTodo = {jest.fn()}
    ></Todo>
    )
  wrapper.find('button.ui.green.button').simulate('click');
  expect(to_json(wrapper)).toMatchSnapshot();
})