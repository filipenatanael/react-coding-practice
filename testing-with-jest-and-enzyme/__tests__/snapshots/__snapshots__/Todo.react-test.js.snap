// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clicking edit button brings up edit form 1`] = `
<div
  className="column"
>
  <div
    className="ui brown card"
  >
    <img
      className="ui image"
      src="https://www.testurl.com/images/image"
    />
    <div
      className="content"
    >
      <div
        className="ui form"
      >
        <div
          className="field"
        >
          <label>
            Title
          </label>
          <input
            defaultValue="Todo"
            name="title"
            onChange={[Function]}
            type="text"
          />
        </div>
        <div
          className="field"
        >
          <label>
            Project
          </label>
          <input
            defaultValue="Project"
            name="project"
            onChange={[Function]}
            type="text"
          />
        </div>
      </div>
    </div>
    <div
      className="extra content"
    >
      <div
        className="ui two buttons"
      >
        <button
          className="ui basic green button"
          onClick={[Function]}
        >
          <i
            className="checkmark icon"
          />
           Update
        </button>
        <button
          className="ui basic red button"
          onClick={[Function]}
        >
          <i
            className="remove icon"
          />
           Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders correctly 1`] = `
<div
  className="column"
>
  <div
    className="ui brown card"
  >
    <img
      className="ui image"
      src="https://www.testurl.com/images/image"
    />
    <div
      className="content"
    >
      <div
        className="header"
      >
        Todo
      </div>
      <div
        className="meta"
      >
        Project
      </div>
      <div
        className="meta"
      >
        Created 
        a day ago
      </div>
    </div>
    <div
      className="extra content"
    >
      <div>
        <div
          className="ui toggle checkbox"
          style={
            Object {
              "marginBottom": "10px",
            }
          }
        >
          <input
            defaultChecked={false}
            name="public"
            onChange={[Function]}
            type="checkbox"
            value="on"
          />
          <label>
            Complete
          </label>
        </div>
        <div
          className="ui two buttons"
        >
          <button
            className="ui basic green button"
            onClick={[Function]}
          >
            Edit
          </button>
          <button
            className="ui basic red button"
            onClick={[Function]}
          >
            Delete
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
